packages/td-tools/src/td-parser.ts

```diff
@@ -222,20 +222,20 @@ export function serializeTD(thing: Thing): string {
     const copy = JSON.parse(JSON.stringify(thing));
 
     // clean-ups
-    if (!copy.security || copy.security.length === 0) {
+    if (copy?.security.length === 0) {
```

Comment:
I also think this proposed change is not correct ... since it is OR not AND

Reply:
fixed with https://github.com/eclipse-thingweb/node-wot/pull/1077/commits/25d5390b5c9213128cd51404367da2bc98f0f6d9